Source/Android/jni/Achievements.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2003 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <jni.h>
+#include <map>
+#include <thread>
+
+#include "Achievements.h"
+
+#include "Common/Event.h"
+#include "Common/Thread.h"
+#include "Common/Logging/Log.h"
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+
+#include "DiscIO/CompressedBlob.h"
+
+// Global java_vm class
+extern JavaVM* g_java_vm;
+
+namespace Achievements
+{
+// Achievement handling function
+static bool InTheCloset(AchievementTypes type)
+{
+	if (Core::GetState() == Core::CORE_UNINITIALIZED || Core::GetState() == Core::CORE_PAUSE)
+		return false;
+
+	if (DiscIO::IsGCZBlob(SConfig::GetInstance().m_strFilename))
```

Comment:
Okay, sorry. IsGCZBlob is still a bit less efficient than checking an already constructed volume, but it's reading so little data that doing it once doesn't matter.

They are just stripped, but there's not really any difference for users. GCZ also relies on scrubbing to make the files smaller - the difference is that it compresses the scrubbed pages instead of removing them entirely, which essentially has the same effect. Of course, the situation is different for GC games, but I don't think anyone uses CISO or WBFS for those anyway.